notification-portlet-webcomponents/notification-component-icon/src/NotificationIcon.js

```diff
@@ -7,52 +7,74 @@ import {
 } from 'reactstrap';
 import FontAwesomeIcon from '@fortawesome/react-fontawesome';
 import {translate} from 'react-i18next';
+import reactTimeout from 'react-timeout';
+import PropTypes from 'prop-types';
 
 class NotificationIcon extends Component {
+  static propTypes = {
+    userInfoApiUrl: PropTypes.string,
+    tokenTimeoutMs: PropTypes.number,
+    notificationApiUrl: PropTypes.string,
+    debug: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    userInfoApiUrl: '/uPortal/api/v5-1/userinfo',
+    tokenTimeoutMs: 180000, // 3 minutes
+    notificationApiUrl: '/NotificationPortlet/api/v2/notifications',
+    debug: false,
+  };
+
   state = {
     isDropdownOpen: false,
-    bearerToken: null,
-    notifications: []
+    bearerToken: 'test', // TODO set to null
+    notifications: [],
   };
 
   fetchBearerToken = async () => {
+    const {setTimeout, userInfoApiUrl, tokenTimeoutMs} = this.props;
+
     try {
-      // TODO: allow override of hard-coded URI
-      const response = await fetch('/uPortal/api/v5-1/userinfo', {
-        credentials: 'same-origin'
+      const response = await fetch(userInfoApiUrl, {
+        credentials: 'same-origin',
```

Comment:
Why a comma here?  Looks wrong -- does ES6 tolerate it?

Reply:
Google code style recommends any object or list should have a comma all items, including the last element.
https://google.github.io/styleguide/jsguide.html#features-arrays-trailing-comma

Trailing commas are allowed by ES6, and babel will handle transpiling for older browser support.